torch/csrc/jit/script/concrete_module_type.h

```diff
@@ -190,12 +193,18 @@ class VISIBILITY_HIDDEN ConcreteModuleType {
   };
 
   struct ModuleInfo {
-    std::string name;
-    std::shared_ptr<ConcreteModuleType> meta;
+    ModuleInfo(std::string name, std::shared_ptr<ConcreteModuleType> meta)
```

Comment:
In the case where this module is actually a module interface, what is meta set to? This seems redundant to me.

Reply:
Yes, we need to be careful about the concepts here. It's not just redundant, the current code is not correct I think. For example, you could use a method that is defined on the `ConcreteModuleType` pyclass, but not in the interface.

This really an either/or relationship, either this is a concrete module instance with a `ConcreteModuleType`, or a module interface that is restricted to the methods defined in the interface. I think we need to split these concepts.

Reply:
I think `ConcreteModuleType` of the submodule here is just used to create the correct JIT type, in the case where the submodule is a module interface, the meta will still be set, but actually  it will be shadowed by the ModuleInterfaceType that we created when we creating the parent JIT class Type (ModuleInfo use type_ as source of truth for creating JIT type). I don't quite understand why we could use a method that is defined on `ConcreteModuleType` pyclass, everything after compilation should use cpp_module as source of truth (and it set the correct ModuleInterface jit type) and concrete_module should be invisible to user?